app/src/batch_process.py

```diff
@@ -11,9 +11,15 @@
 
 
 async def batch_process(file_path: str, engine: ChatEngineInterface) -> str:
-    with open(file_path, mode="r", newline="", encoding="utf-8") as csvfile:
+
+    # Using utf-8-sig to handle CSV files with BOM characters from Excel
+    with open(file_path, mode="r", newline="", encoding="utf-8-sig") as csvfile:
         reader = csv.DictReader(csvfile)
 
+        # Clean up fieldnames to remove any BOM characters
+        if reader.fieldnames:
+            reader.fieldnames = [f.strip("\ufeff") for f in reader.fieldnames]
```

Comment:
This line bothers me since it's specifically for Excel-exported files. How about this approach instead: Re-encode file contents into `utf-8` before reading it with `DictReader` -- see https://stackoverflow.com/a/8898439/23458508. This makes it more obvious and compartmentalized, and you can add a code comment for the re-encoding code block that it's for Excel.

Reply:
Yes! Good idea!